Assignment of Akshay Naik - Newsboard App

akshaynaik404 opened the pull request:
Link to App: [https://newsboard.tk](https://newsboard.tk)

akshaynaik404 committed:
Revert CRA project back to its original state and add yarn.lock

akshaynaik404 committed:
Intial Commit of the project

* Add necessary packages to package.json & Update yarn.lock
* Remove margin on body in index.html
* Remove unnecessary files (css, svg)
* Add Newsboard related code - Currently showing latest news from
  The Next Web

akshaynaik404 committed:
Add Firebase related config files

akshaynaik404 committed:
[docs] Update README.md - Add Project Documentation

  * Also add npm deploy script in package.json

akshaynaik404 committed:
[NewsItems] Refactor PostList

  * Rename PostList to NewsItems
  * Add Loader and ListWithDividers Components
  * and Update README.md to check 2nd assignment assignment requirement

akshaynaik404 committed:
Update theme_color

akshaynaik404 commented:
Web Application is live at [Newsboard](https://newsboard.tk).

upa8 commented:
@akshaynaik404  

Good Job ! ... !!! We will get back to you once we review it.

akshaynaik404 committed:
Add cross-env support in npm scripts to for better import paths

  * Add cross-env to package.json
  * Update yarn.lock

akshaynaik404 committed:
Move api to its own directory inside root

akshaynaik404 committed:
[NewsItems] Refactor to add Network Error Dialog

akshaynaik404 committed:
[App] Refactor App and NewsItems component

  - Move api call, other state properties from NewsItems
    component to App component.

akshaynaik404 committed:
[ListWithDividers] Make News titles clickable

akshaynaik404 committed:
[FabBtn] Add FabBtn component and clean App.js code

akshaynaik404 commented:
[This](https://github.com/javascripters-community/thatjsdude/pull/4/commits/1034ef0ae858aebaf442fc6248a0583aba2c6da3) Commit will remove the code labs error.

akshaynaik404 committed:
Add Fork on Github ribbon at the top right corner

akshaynaik404 committed:
[App] Add proper styling
  * Center AppBar headline
  * Set maxWidth on ListWithDividers
  * Style FabBtn to bottom center

akshaynaik404 committed:
Add lint support

- Add eslint and its peer dependencies, lint script in package.json
- Change line endings from CRLF to LF

akshaynaik404 committed:
Add necessary eslint config files and fix lint errors.

akshaynaik404 committed:
Fix last news item being overlapped by cta.

akshaynaik404 committed:
Update dependencies as per eslint-config-airbnb README.md

javascripters2015 reviewed (changes requested):
Please update README file with all the required instructions to setup the project for any new person.
Add comments whereever required in code to understand the intention of code implimentation
Avoid inline css and js

Rest of the code quality looks good to us.
Good Job.. keep it up.

akshaynaik404 commented:
Thanks for the review.

Although README.md already contains the installation instructions

akshaynaik404 commented:
Can you elaborate on inline css, js point?

javascripters2015 commented on public/index.html:
```diff
@@ -19,13 +19,20 @@
+    <title>Newsboard</title>
   </head>
-  <body>
+  <body style="margin: 0">
```
Avoid inline css

akshaynaik404 replied:
Ahh! Yes.  That one. I wrote it inline to avoid creating another styles.css file just for this one rule.

javascripters2015 commented on src/api/index.js:
```diff
@@ -0,0 +1,7 @@
+export default {
+  getNewsItems() {
+    return axios.get('https://newsapi.org/v1/articles?source=the-next-web&sortBy=latest&apiKey=14d7c1ccdc5042c9b70b41aebde1348a');
+  },
```
Avoid unnecessary comma

akshaynaik404 replied:
The trailing comma helps in achieving clarity in diffs. [Eslint rule](http://eslint.org/docs/rules/comma-dangle)

javascripters2015 commented on src/components/FabBtn/index.js:
```diff
@@ -0,0 +1,45 @@
+    width: '100%',
+    bottom: 16,
+    textAlign: 'center',
+  },
```
Avoid unnecessary comma

javascripters2015 reviewed (commented):
Please avoid unnecessary ending comma

akshaynaik404 commented on README.md:
```diff
@@ -1,49 +1,2172 @@
 
-from terminal using commands:   https://www.atlassian.com/git/tutorials/learn-git-with-bitbucket-cloud 
+1. Clone this repo.
+2. Execute `npm install` or `yarn` on your terminal.
```
CD into cloned repo directory and then run yarn.
